src/sctools/gtf.py

```diff
@@ -244,3 +244,44 @@ def filter(self, retain_types: Iterable[str]) -> Generator:
         for record in self:
             if record.feature in retain_types:
                 yield record
+
+
+# todo this lenient behavior is deemed to change in the future (warning -> exception)
+def _resolve_multiple_gene_names(gene_name: str):
+    _logger.warning(f'Multiple entries encountered for "{gene_name}". Please validate the input GTF file(s). '
+                    f'Skipping the record for now; in the future, this will be considered as a '
+                    f'malformed GTF file.')
+
+
+def extract_gene_names(files='-', mode='r', header_comment_char='#') -> Dict[str, int]:
```

Comment:
What are your feeling on types for parameters with default arguments? Mypy suggests it because it enforces static typing on them (in addition to annotating them as strings, which is obvious given the default). I think I like this, but I'm open to opinions.

Reply:
Added types.